addons/pos_sale/tests/test_pos_sale_flow.py

```diff
@@ -622,3 +623,42 @@ def test_downpayment_amount_to_invoice(self):
         # Confirm all invoices
         sale_order.invoice_ids.action_post()
         self.assertEqual(sale_order.order_line[1].price_unit, 10)
+
+    def test_downpayment_line_name(self):
+        product_a = self.env['product.product'].create({
+            'name': 'Product A',
+            'available_in_pos': True,
+            'type': 'product',
+            'lst_price': 100.0,
+            'taxes_id': [],
+        })
+        partner_test = self.env['res.partner'].create({'name': 'Test Partner'})
+
+        sale_order = self.env['sale.order'].create({
+            'partner_id': partner_test.id,
+            'order_line': [(0, 0, {
+                'product_id': product_a.id,
+                'name': product_a.name,
+                'product_uom_qty': 1,
+                'product_uom': product_a.uom_id.id,
+                'price_unit': product_a.lst_price,
+            })],
+        })
+        sale_order.action_confirm()
+
+        self.downpayment_product = self.env['product.product'].create({
+            'name': 'Down Payment',
+            'available_in_pos': True,
+            'type': 'service',
+            'taxes_id': [],
+        })
```

Comment:
```suggestion
        downpayment_product = self.env['product.product'].create({
            'name': 'Down Payment',
            'available_in_pos': True,
            'type': 'service',
            'taxes_id': [],
        })
```
Don't assign a variable to `self` if only needed in the current test.

Reply:
Indeed, not needed elsewhere